src/network/networkd-dhcp4.c

```
@@ -1141,6 +1144,219 @@ static int dhcp_lease_acquired(sd_dhcp_client *client, Link *link) {
+        return 1;
+}
+
+static int dhcp_configure_with_persistent_lease(Link* link) {
+
+        _cleanup_free_ char *lease_file = NULL;
+        int dir_fd;
+        int r;
+
+        r = link_get_dhcp_client_lease_path(link, &dir_fd, &lease_file);
+
+        if (r < 0) {
+                log_link_debug_errno(link, r, "Failed to get persistent lease path, ignoring: %m");
+                return r;
+        }
+
+        if (r > 0) {
+                r = sd_dhcp_client_set_lease_file(link->dhcp_client, dir_fd, lease_file);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to set lease file: %m");
+
+                /*load persistent lease, before restarting client */
+                r = dhcp4_load_persistent_lease(link);
+
+                if (r > 0) {
+                        link->dhcp4_configured = false; // mark link for configuration
+
+                        r = dhcp4_request_address_and_routes(link, true);
+                        if (r < 0)
+                                log_link_warning_errno(link, r, "Failed to configure loaded lease: %m");
+
+                } else if (r < 0) {
+                        log_link_warning_errno(link, r, "Failed to load persistent lease: %m");
+                }
+        }
+
+        return 1;
+}
+
+void manager_enable_dhcp4_client_persistent_storage(Manager *manager, bool start) {
+        Link *link;
+        int r;
+
+        assert(manager);
+
+        /* Keep the flag that the storage is ready for use */
+
+        manager->persistent_storage_ready = start;
+
+        HASHMAP_FOREACH(link, manager->links_by_index) {
+                if (!link->dhcp_client)
+                        continue;
+
+                if(!dhcp_client_persist_leases(link))
+                        continue;
+
+                r = sd_dhcp_client_stop(link->dhcp_client);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to stop DHCP client: %m");
```

If the failure is ignored. Please mention that so in the log message.
```c
"Failed to stop DHCP client, ignoring: %m"
```